### ISSUE TITLE
Something wrong how RSI is calculated

### ISSUE THREAD

**Issue by Krisa** [sdcoffey/techan#13]
I can't say for sure what's wrong (I'm not familiar how RSI is calculated), but the calculation itself is not providing the right result when I compare to what I see on Tradingview. To triple-check I was not doing something wrong, I tried with this library https://github.com/markcheno/go-talib, which provided the expected result. 

Also I'm using successfully the EMA of this library, which works well, so I highly suspect that the RSI calculation is not correct.

**Comment by sdcoffey** [comment]
Hey @Krisa,

sorry for the delayed response here. Can you provide an example?

**Comment by Krisa** [comment]
@sdcoffey 

here: 
```
package main

import (
	"context"
	"fmt"
	"log"
	"time"

	"github.com/adshao/go-binance"
	"github.com/sdcoffey/big"
	"github.com/sdcoffey/techan"
)

func main() {
	series := getSeries()
	closePrices := techan.NewClosePriceIndicator(series)

	// Show EMAs
	EMA10 := techan.NewEMAIndicator(closePrices, 10)
	EMA20 := techan.NewEMAIndicator(closePrices, 20)
	EMA10Price := EMA10.Calculate(series.LastIndex() - 1).Float()
	EMA20Price := EMA20.Calculate(series.LastIndex() - 1).Float()

	log.Printf("EMA10Price: %v", EMA10Price)
	log.Printf("EMA20Price: %v", EMA20Price)

	// Show RSI
	rsi14 := techan.NewRelativeStrengthIndexIndicator(closePrices, 14)
	rsi14Calculated := rsi14.Calculate(series.LastIndex() - 1).Float()

	log.Printf("rsi14: %v", rsi14Calculated)
}

func getSeries() *techan.TimeSeries {
	binanceClient := binance.NewClient("", "")
	klines, err := binanceClient.
		NewKlinesService().
		Symbol("BTCUSDT").
		Interval("1h").
		Do(context.Background())

	if err != nil {
		fmt.Println(err)
		return nil
	}

	series := techan.NewTimeSeries()

	for _, kline := range klines {
		start := time.Unix(0, 1000000*kline.OpenTime)
		end := time.Unix(0, 1000000*kline.CloseTime)
		period := techan.NewTimePeriod(start, end.Add(1*time.Millisecond).Sub(start))

		candle := techan.NewCandle(period)
		candle.OpenPrice = big.NewFromString(kline.Open)
		candle.ClosePrice = big.NewFromString(kline.Close)
		candle.MaxPrice = big.NewFromString(kline.High)
		candle.MinPrice = big.NewFromString(kline.Low)
		candle.Volume = big.NewFromString(kline.Volume)
		candle.TradeCount = uint(kline.TradeNum)

		series.AddCandle(candle)
	}
	return series
}
```

You can compare to the real chart here: https://www.tradingview.com/symbols/BTCUSDT/?exchange=BINANCE

EMA 10 and 20 are both perfectly calculated, RSI, at the time of writing (since it takes `time.Now() - 1 hour`) is quite off (45.53 real, 51.26 calculated)

**Comment by sdcoffey** [comment]
Thanks for the snippet @Krisa, let me see if I can fix

**Comment by sdcoffey** [comment]
@Krisa sorry for taking so long to get to this. The issue should be fixed in v0.8.0, please give it a whirl and re-open if it persists.
